autopush_rs/src/protocol.rs

```diff
@@ -120,6 +123,25 @@ pub struct Notification {
     pub headers: Option<HashMap<String, String>>,
 }
 
+impl Notification {
+    /// Generate a sort key suitable for use as the chidmessageid
+    pub fn sort_key(&self) -> String {
+        if let Some(ref topic) = self.topic {
+            format!("01:{}:{}", self.channel_id.hyphenated(), topic)
+        } else if let Some(ref sortkey_timestamp) = self.sortkey_timestamp {
```

Comment:
u64 is Copy so this could be a lot simpler. probably like

```rust
if let Some(sortkey_timestamp) = ... {
    format!(
        "..",
        if sortkey_timestamp == 0 { ms_since_epoch } else { sortkey_timestamp },
        ...
    )
}
```